src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -21,148 +87,1634 @@ const ActorInit En_Ossan_InitVars = {
+    if (CHECK_BTN_ALL(input->press.button, BTN_B)) {
+        this->actionFunc = this->tmpActionFunc;
+        func_80151938(globalCtx, this->items[this->cursorIdx]->actor.textId);
+        return true;
+    }
+    return false;
+}
+
+void EnOssan_SetupStartShopping(GlobalContext* globalCtx, EnOssan* this, u8 skipHello) {
+    func_8011552C(globalCtx, 0x10);
+    if (!skipHello) {
+        EnOssan_SetupAction(this, EnOssan_Hello);
+    } else {
+        EnOssan_StartShopping(globalCtx, this);
+    }
+}
+
+void EnOssan_StartShopping(GlobalContext* globalCtx, EnOssan* this) {
+    EnOssan_SetupAction(this, EnOssan_FaceShopkeeper);
+    func_80151938(globalCtx, 0x0640);
+    func_8011552C(globalCtx, 6);
+    this->stickRightPrompt.isEnabled = true;
+    this->stickLeftPrompt.isEnabled = true;
+}
+
+void EnOssan_SetupLookToShopkeeperFromShelf(GlobalContext* globalCtx, EnOssan* this) {
+    play_sound(NA_SE_SY_CURSOR);
+    this->drawCursor = 0;
+    EnOssan_SetupAction(this, EnOssan_LookToShopkeeperFromShelf);
+}
+
+void EnOssan_Idle(EnOssan* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    func_8013D9C8(globalCtx, this->limbRotTableY, this->limbRotTableZ, 19);
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        player->stateFlags2 |= 0x20000000;
+        EnOssan_SetupAction(this, EnOssan_BeginInteraction);
+        if (this->cutsceneState == 0) {
+            if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+                ActorCutscene_Stop(0x7C);
+            }
+            this->cutscene = this->lookToShopkeeperCutscene;
+            ActorCutscene_SetIntentToPlay(this->cutscene);
+            this->cutsceneState = 1;
+        }
+    } else {
+        if (this->actor.xzDistToPlayer < 100.0f &&
+            (player->actor.world.pos.x >= -40.0f && player->actor.world.pos.x <= 40.0f) &&
+            (player->actor.world.pos.z >= -91.0f && player->actor.world.pos.z <= -60.0f)) {
+            func_800B8614(&this->actor, globalCtx, 100.0f);
+        }
+        if (this->actor.params == ENOSSAN_PART_TIME_WORKER) {
+            Math_SmoothStepToS(&this->headRotPartTimeWorker.y, 8000, 3, 2000, 0);
+        }
+    }
+}
+
+void EnOssan_BeginInteraction(EnOssan* this, GlobalContext* globalCtx) {
+    ActorAnimationEntryS* animations = sAnimations[this->actor.params];
+    s16 curFrame = this->skelAnime.animCurrentFrame;
+    s16 frameCount;
+
+    frameCount = SkelAnime_GetFrameCount(&animations[this->animationIdx].animationSeg->common);
+    if (this->animationIdx == 3) {
+        frameCount = 0;
+    }
+    if (this->cutsceneState == 1) {
+        if (ActorCutscene_GetCanPlayNext(this->cutscene)) {
+            ActorCutscene_StartAndSetFlag(this->cutscene, &this->actor);
+            this->cutsceneState = 2;
+        } else {
+            ActorCutscene_SetIntentToPlay(this->cutscene);
+        }
+    }
+    if (this->actor.params == ENOSSAN_CURIOSITY_SHOP_MAN) {
+        if (curFrame == frameCount) {
```

Comment:
That is actually a different logic flow. Since the else would be reached if  `this->actor.params != ENOSSAN_CURIOSITY_SHOP_MAN` OR `curFrame != frameCount`.